pkg/common/cns-lib/node/manager.go

```diff
@@ -129,16 +131,18 @@ func (m *defaultManager) DiscoverNode(ctx context.Context, nodeUUID string) erro
 func (m *defaultManager) GetNodeByName(ctx context.Context, nodeName string) (*vsphere.VirtualMachine, error) {
 	log := logger.GetLogger(ctx)
 	nodeUUID, found := m.nodeNameToUUID.Load(nodeName)
-	if !found {
-		log.Errorf("Node not found with nodeName %s", nodeName)
-		return nil, ErrNodeNotFound
-	}
-	if nodeUUID != nil && nodeUUID.(string) != "" {
-		return m.GetNode(ctx, nodeUUID.(string), nil)
+	if found {
+		if nodeUUID.(string) != "" {
+			return m.GetNode(ctx, nodeUUID.(string), nil)
+		}
+		log.Infof("Empty nodeUUID observed in cache for the node: %q", nodeName)
```

Comment:
what is the main reason to change the logic here?
Previously, if nodeName is not found from nodeNameToUUID map, it returns not found error.
Now, it will continue to check k8s client to see whether it can find there?

What is the difference between these two places to get node uuid?

Reply:
The `nodeNameToUUID` map is filled by a listener to an informer (see https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/vanilla/nodes.go#L53). In rare cases this map may not be quite up to date. As the `ErrNodeNotFound` error can now have a side effect (volume detach), it is better for  robustness to make sure the node is really not existing.

Reply:
got it, thanks!
I have another question related to this function https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/529/files#r594785655